cmd/did-method-cli/createdidcmd/createdid.go

```diff
@@ -0,0 +1,235 @@
+		" Alternatively, this can be set with the following environment variable: " + tlsSystemCertPoolEnvKey
+	tlsSystemCertPoolEnvKey = "DID_METHOD_CLI_TLS_SYSTEMCERTPOOL"
+
+	tlsCACertsFlagName  = "tls-cacerts"
+	tlsCACertsFlagUsage = "Comma-Separated list of ca certs path." +
+		" Alternatively, this can be set with the following environment variable: " + tlsCACertsEnvKey
+	tlsCACertsEnvKey = "DID_METHOD_CLI_TLS_CACERTS"
+
+	sidetreeWriteTokenFlagName  = "sidetree-write-token"
+	sidetreeWriteTokenEnvKey    = "DID_METHOD_CLI_SIDETREE_WRITE_TOKEN" //nolint: gosec
+	sidetreeWriteTokenFlagUsage = "The sidetree write token " +
+		" Alternatively, this can be set with the following environment variable: " + sidetreeWriteTokenEnvKey
+
+	publicKeyFileFlagName  = "publickey-file"
+	publicKeyFileEnvKey    = "DID_METHOD_CLI_PUBLICKEY_FILE"
+	publicKeyFileFlagUsage = "publickey file include public keys for Trustbloc DID " +
+		" Alternatively, this can be set with the following environment variable: " + publicKeyFileEnvKey
+
+	serviceFlagName   = "service-file"
+	serviceFileEnvKey = "DID_METHOD_CLI_SERVICE_FILE"
+	serviceFlagUsage  = "publickey file include services for Trustbloc DID " +
+		" Alternatively, this can be set with the following environment variable: " + serviceFileEnvKey
+)
+
+// GetCreateDIDCmd returns the Cobra create did command.
+func GetCreateDIDCmd() *cobra.Command {
+	createDIDCmd := createDIDCmd()
+
+	createFlags(createDIDCmd)
+
+	return createDIDCmd
+}
+
+func createDIDCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:   "create-did",
+		Short: "Create TrustBloc DID",
+		Long:  "Create TrustBloc DID",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			rootCAs, err := getRootCAs(cmd)
+			if err != nil {
+				return err
+			}
+
+			sidetreeWriteToken, err := cmdutils.GetUserSetVarFromString(cmd, sidetreeWriteTokenFlagName,
+				sidetreeWriteTokenEnvKey, true)
+			if err != nil {
+				return err
+			}
+
+			client := did.New(did.WithAuthToken(sidetreeWriteToken),
+				did.WithTLSConfig(&tls.Config{RootCAs: rootCAs}))
+
+			domain, err := cmdutils.GetUserSetVarFromString(cmd, domainFlagName,
+				domainFileEnvKey, false)
+			if err != nil {
+				return err
+			}
+
+			opts, err := getPublicKeys(cmd)
+			if err != nil {
+				return err
+			}
+
+			serviceOpts, err := getServices(cmd)
+			if err != nil {
+				return err
+			}
+
+			opts = append(opts, serviceOpts...)
+
+			doc, err := client.CreateDID(domain, opts...)
+			if err != nil {
+				return err
+			}
+
+			bytes, err := doc.JSONBytes()
+			if err != nil {
+				return err
+			}
+
+			fmt.Println(string(bytes))
+
+			return nil
+		},
+	}
+}
+
+func getServices(cmd *cobra.Command) ([]did.CreateDIDOption, error) {
+	serviceFile, err := cmdutils.GetUserSetVarFromString(cmd, serviceFlagName,
+		serviceFileEnvKey, false)
+	if err != nil {
+		return nil, err
+	}
+
+	svcData, err := ioutil.ReadFile(serviceFile) //nolint: gosec
```

Comment:
Is is practical for user to provide two files (one for services and one for public keys) in order to create did? This might be ok @troyronda

Reply:
Seems odd to me.